Patch by Robert Scheck <robert@fedoraproject.org> for z-push >= 2.1.1, which changes the original
paths from Z-Push upstream to Fedora compliant ones. Additionally, it adds some installation and
configuration hints regarding multiple data backend providers.

--- z-push-2.1.1-1788/config.php				2013-11-06 19:20:31.000000000 +0100
+++ z-push-2.1.1-1788/config.php.package			2013-12-12 16:19:30.000000000 +0100
@@ -60,6 +60,7 @@
     // This setting specifies the owner parameter in the certificate to look at.
     define("CERTIFICATE_OWNER_PARAMETER", "SSL_CLIENT_S_DN_CN");
 
+
 /**********************************************************************************
  *  Default FileStateMachine settings
  */
@@ -101,6 +102,7 @@
     // Uncomment and modify the following line if the validation of the certificates fails.
     // define('CAINFO', '/etc/ssl/certs/EmailCA.pem');
 
+
 /**********************************************************************************
  *  Mobile settings
  */
@@ -185,19 +187,33 @@
     // default: 49152 - 48 KB default max photo size in bytes
     define('SYNC_CONTACTS_MAXPICTURESIZE', 49152);
 
+
 /**********************************************************************************
  *  Backend settings
+ *
+ *  Z-Push supports currently the data backend providers Zarafa, IMAP, vCard directory,
+ *  Maildir and a mixture by the combined one. Ensure that the used backend is installed
+ *  and proper configured as well:
+ *    BackendZarafa   - configure /etc/z-push/zarafa.php (provided by z-push-zarafa)
+ *    BackendCombined - configure /etc/z-push/combined.php (provided by z-push-combined)
+ *                      and of course the wanted (at least two) other backend providers
+ *    BackendIMAP     - configure /etc/z-push/imap.php (provided by z-push-imap)
+ *    BackendVCardDir - configure /etc/z-push/vcarddir.php (provided by z-push-vcarddir)
+ *    BackendMaildir  - configure /etc/z-push/maildir.php (provided by z-push-maildir)
+ *  If only one data backend provider has been installed, that one is used by default.
+ *  In case of multiple installed data backend providers, they are tried in order above.
  */
-    // the backend data provider
+    // The backend data provider
     define('BACKEND_PROVIDER', '');
 
+
 /**********************************************************************************
  *  Search provider settings
  *
  *  Alternative backend to perform SEARCH requests (GAL search)
  *  By default the main Backend defines the preferred search functionality.
  *  If set, the Search Provider will always be preferred.
- *  Use 'BackendSearchLDAP' to search in a LDAP directory (see backend/searchldap/config.php)
+ *  Use 'BackendSearchLDAP' to search in a LDAP directory (see /etc/z-push/searchldap.php)
  */
     define('SEARCH_PROVIDER', '');
     // Time in seconds for the server search. Setting it too high might result in timeout.
--- z-push-2.1.1-1788/z-push-admin.php				2013-02-28 22:02:27.000000000 +0100
+++ z-push-2.1.1-1788/z-push-admin.php.package			2013-12-12 16:20:55.000000000 +0100
@@ -44,6 +44,10 @@
 * Consult LICENSE file for details
 ************************************************/
 
+// Script stays in /usr/sbin and is not a symlink to /usr/share/z-push
+define('BASE_PATH_CLI', "/usr/share/z-push/");
+set_include_path(get_include_path() . PATH_SEPARATOR . BASE_PATH_CLI);
+
 include('lib/core/zpushdefs.php');
 include('lib/core/zpush.php');
 include('lib/core/stateobject.php');
@@ -102,8 +106,6 @@
 /************************************************
  * MAIN
  */
-    define('BASE_PATH_CLI',  dirname(__FILE__) ."/");
-    set_include_path(get_include_path() . PATH_SEPARATOR . BASE_PATH_CLI);
     try {
         ZPush::CheckConfig();
         ZPushAdminCLI::CheckEnv();
--- z-push-2.1.1-1788/z-push-top.php				2013-01-04 17:51:29.000000000 +0100
+++ z-push-2.1.1-1788/z-push-top.php.package			2013-12-12 16:21:28.000000000 +0100
@@ -44,6 +44,10 @@
 * Consult LICENSE file for details
 ************************************************/
 
+// Script stays in /usr/sbin and is not a symlink to /usr/share/z-push
+define('BASE_PATH_CLI', "/usr/share/z-push/");
+set_include_path(get_include_path() . PATH_SEPARATOR . BASE_PATH_CLI);
+
 include('lib/exceptions/exceptions.php');
 include('lib/core/zpushdefs.php');
 include('lib/core/zpush.php');
@@ -60,7 +64,6 @@
  * MAIN
  */
     declare(ticks = 1);
-    define('BASE_PATH_CLI',  dirname(__FILE__) ."/");
 
     try {
         ZPush::CheckConfig();
--- z-push-2.1.1-1788/tools/migrate-2.0.x-2.1.0.php		2013-09-05 20:50:34.000000000 +0200
+++ z-push-2.1.1-1788/tools/migrate-2.0.x-2.1.0.php.package	2013-12-12 22:04:34.000000000 +0100
@@ -44,7 +44,7 @@
 ************************************************/
 
 // Please adjust to match your z-push installation directory, usually /usr/share/z-push
-define('ZPUSH_BASE_PATH', "../");
+define('ZPUSH_BASE_PATH', "/usr/share/z-push/");
 
 
 
